src/components/InventoryGroups/Modals/DeleteGroupModal.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import validatorTypes from '@data-driven-forms/react-form-renderer/validator-types';
+import componentTypes from '@data-driven-forms/react-form-renderer/component-types';
+import Modal from './Modal';
+import { deleteGroupById } from '../utils/api';
+import { ExclamationTriangleIcon } from '@patternfly/react-icons/dist/esm/icons/exclamation-triangle-icon';
+import warningColor from '@patternfly/react-tokens/dist/esm/global_warning_color_100';
+import { Text } from '@patternfly/react-core';
+import apiWithToast from '../utils/apiWithToast';
+import { useDispatch } from 'react-redux';
+
+const description = (name) => (
+    <Text>
+        <strong>{name} </strong>and all its data will be permanently deleted.
+    Associated systems will be removed from the group but will not be deleted.
+    </Text>
+);
+
+const schema = (name) => ({
+    fields: [
+        {
+            component: componentTypes.PLAIN_TEXT,
+            name: 'warning-message',
+            label: description(name)
+        },
+        {
+            component: componentTypes.CHECKBOX,
+            name: 'confirmation',
+            label: 'I understand that this action cannot be undone.',
+            validate: [{ type: validatorTypes.REQUIRED }]
+        }
+    ]
+});
+
+const WarningIcon = () => (
+    <ExclamationTriangleIcon color={warningColor.value} />
+);
+
```

Comment:
nit-pick: don't define another wrapper and just feed ExclamationTriangleIcon to the props below to avoid extra code.

Reply:
You are correct this is much better